### PR TITLE
Change `plots` submodule name to `interpret`

### PR DESCRIPTION
**Pull request by GStechschulte**
This PR changes the `plots` submodule name to `interpret` as the functions in this sub package are not limited to plotting functions only. Additionally, this draft PR aims exposes the sub package functions in the main Bambi package to allow imports such as

```python3
import bambi as bmb

bmb.interpret.predictions()
# or
bmb.interpret.plot_predictions()
```

Any referencing files, documentation, and tests have been edited to reflect this change.

_Note_: this branch was a checkout of the branch for PR #704 (once PR 704 is merged, then this can be merged).

### CODE REVIEW

File: docs/notebooks/plot_comparisons.ipynb [hunk 1]
```diff
@@ -7,7 +7,7 @@
...
+    "`comparisons` and `plot_comparisons` are a part of Bambi's sub-package `interprets` that feature a set of functions used to interpret complex regression models. This sub-package is inspired by the R package [marginaleffects](https://vincentarelbundock.github.io/marginaleffects/articles/predictions.html#conditional-adjusted-predictions-plot). These two functions allow the modeler to **compare** the predictions made by a model for different contrast and covariate values. Below, it is described why comparing predictions is useful in interpreting generalized linear models (GLMs), how this methodology is implemented in Bambi, and how to use `comparisons` and `plot_comparisons`. It is assumed that the reader is familiar with the basics of GLMs. If not, refer to the Bambi [Basic Building Blocks](https://bambinos.github.io/bambi/notebooks/how_bambi_works.html#Link-functions) example.\n",
```
Comment (aloctavodia) [inline docs/notebooks/plot_comparisons.ipynb:10]:
```suggestion
    "`comparisons` and `plot_comparisons` are a part of Bambi's sub-package `interpret` that feature a set of functions used to interpret complex regression models. This sub-package is inspired by the R package [marginaleffects](https://vincentarelbundock.github.io/marginaleffects/articles/predictions.html#conditional-adjusted-predictions-plot). These two functions allow the modeler to **compare** the predictions made by a model for different contrast and covariate values. Below, it is described why comparing predictions is useful in interpreting generalized linear models (GLMs), how this methodology is implemented in Bambi, and how to use `comparisons` and `plot_comparisons`. It is assumed that the reader is familiar with the basics of GLMs. If not, refer to the Bambi [Basic Building Blocks](https://bambinos.github.io/bambi/notebooks/how_bambi_works.html#Link-functions) example.\n",
```

**Comment by aloctavodia** [comment]
I think the errors come from this line 

https://github.com/bambinos/bambi/blob/main/pyproject.toml#L61C20-L61C20

as that directory no longer exists

**Review by aloctavodia — APPROVED** [review]

**Comment by tomicapretto** [comment]
@GStechschulte if I merge this, there's no need to merge #704, right?

**Comment by GStechschulte** [comment]
> @GStechschulte if I merge this, there's no need to merge #704, right?
> 
> **edit** I merged and created conflicts 😅 , I will fix them

Just let me know if I should do something. It is also my fault. I created excess work with PR #704. It could have all been in one.

**Comment by tomicapretto** [comment]
> > @GStechschulte if I merge this, there's no need to merge #704, right?
> > **edit** I merged and created conflicts 😅 , I will fix them
> 
> Just let me know if I should do something. It is also my fault. I created excess work with PR #704 since it could have all been in one PR.

no worries! It's fixed now

**Pull request merged**